control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp

```diff
@@ -520,6 +522,13 @@ AckermannControlCommand VehicleCmdGate::filterControlCommand(const AckermannCont
   // filter in manual mode.
   prev_values.longitudinal = out.longitudinal;  // TODO(Horibe): to be removed
 
+  // When ego is stopped, consider the actual vehicle longitudinal state for the next filtering
+  // this is to prevent the jerk limits being applied on the "stop acceleration",
+  // which may be negative and cause delays when restarting the vehicle.
+  if (ego_is_stopped) {
+    prev_values.longitudinal = current_status_cmd.longitudinal;
+  }
```

Comment:
@maxime-clem I found a concern. Assume that the vehicle will not move with a small target velocity (e.g. 1.0m/s).
In this case, if `prev_values` is set with a current velocity (or acceleration) every time when the vehicle is stopped, the output target velocity will not exceed `v_current + a_lim * dt` (which is usually very small) as long as the vehicle is stopped.

If the vehicle will not move with the target velocity `v_current + a_lim * dt`, then the vehicle will never start moving.

So, I suggest changing the logic like this:

```
bool hasSmallValue(cmd) {
    if (cmd.acceleration < -0.5) {
        true;
    }
    return false;
}

// not only check the stop status but also check if the current acceleration is large to negative direction
if (ego_is_stopped && hasSmallValue(prev_values)) {
    prev_values.longitudinal = current_status_cmd.longitudinal;
}
```

Reply:
Thank you for finding this issue.
Your solution sounds good. I implemented it by adding a condition:
```
const auto input_cmd_is_stopping = in.longitudinal.acceleration < 0.0;
```

Reply:
Looks good!